javascript/ql/src/semmle/javascript/dataflow/internal/FlowSteps.qll

```diff
@@ -125,20 +125,27 @@ private module CachedSteps {
    */
   cached
   predicate argumentPassing(
-    DataFlow::InvokeNode invk, DataFlow::ValueNode arg, Function f, DataFlow::ParameterNode parm
+    DataFlow::InvokeNode invk, DataFlow::ValueNode arg, Function f, DataFlow::SourceNode parm
   ) {
     calls(invk, f) and
-    exists(int i |
-      f.getParameter(i) = parm.getParameter() and
-      not parm.isRestParameter() and
-      arg = invk.getArgument(i)
+    (
+      exists(int i, Parameter p |
+        f.getParameter(i) = p and
+        not p.isRestParameter() and
+        arg = invk.getArgument(i) and
+        parm = DataFlow::parameterNode(p)
+      )
+      or
+      arg = invk.(DataFlow::MethodCallNode).getReceiver() and
```

Comment:
If we use `invk.(DataFlow::CallNode).getReceiver()` here we'll also handle receivers passed in through reflective calls.

Reply:
Ah, good point. This was simply a typo, I think. I'll do a quick sanity check to verify that this doesn't upset the optimiser.

Reply:
Sanity check showed no performance regression.